### DISCUSSION TITLE
Any suggestion ? I got 'mqtt untouched' message  , and  mqtt topic not publish

### DISCUSSION THREAD

**Discussion by naphanz** [rednblkx/HomeKey-ESP32#34] [Q&A]
I got 'mqtt untouched' message  , and  mqtt topic that represent status of scan by applekey not publish

![image](https://github.com/rednblkx/HomeKey-ESP32/assets/96221576/e517238f-6908-4e2e-a8cf-9e26e54bf0d8)
from vscode serial monitor
![image](https://github.com/rednblkx/HomeKey-ESP32/assets/96221576/7492d181-26b3-4099-9627-cd0bca576b61)
but when power esp32 discovery messages appeared normally
from mqtt explorer

Thank You in Advance

**Comment by rednblkx** [comment]
Hi there!

It looks like it failed to authenticate the HomeKey. 
Have you successfully finished the setup on the Apple Home App?
What NFC module are you using? If you have the square red one from elechouse know that there are clones that use cheap  resistors which cause issues.
Please set the log level to verbose with the command `@LV` and make another attempt, you can email me the log to rednblkx@protonmail.com.

**Comment by naphanz** [comment]
Have you successfully finished the setup on the Apple Home App? Yes
What NFC module are you using? PN532 5$ from ali T^T 
*I have read ur notice about this before ,but  I don't have much budget

**Reply by rednblkx** [reply]
Looking at the logs, you are missing the lock's private key that you get during setup. Try to re-add it in your Home or just close the Home app and open it again, that should trigger it.

**Reply by rednblkx** [reply]
There is unfortunately nothing much that i can do about the missing key as that is provisioned during setup by the HAP Controller.

**Comment by rednblkx** [comment]
It should now work, please check the new code.

**Comment by naphanz** [comment]
it work now, Thx.
